detectors/resources/src/main/java/com/google/cloud/opentelemetry/detectors/GAEResource.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2022 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.opentelemetry.detectors;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.autoconfigure.spi.ResourceProvider;
+import io.opentelemetry.sdk.resources.Resource;
+import io.opentelemetry.semconv.resource.attributes.ResourceAttributes;
+
+public class GAEResource implements ResourceProvider {
+  private static final String GAE_ENVIRONMENT = "GAE_ENV";
+  private final GCPMetadataConfig metadata;
+  private final EnvVars envVars;
+
+  public GAEResource() {
+    this.metadata = GCPMetadataConfig.DEFAULT_INSTANCE;
+    this.envVars = EnvVars.DEFAULT_INSTANCE;
+  }
+
+  // For testing only
+  GAEResource(GCPMetadataConfig metadata, EnvVars envVars) {
+    this.metadata = metadata;
+    this.envVars = envVars;
+  }
+
+  public Attributes getAttributes() {
+    if (!metadata.isRunningOnGcp()) {
+      return Attributes.empty();
+    }
+
+    AttributesBuilder attrBuilders = Attributes.builder();
+    attrBuilders.put(ResourceAttributes.CLOUD_PROVIDER, ResourceAttributes.CloudProviderValues.GCP);
```

Comment:
If we used the GAEResource processor along with other processors, would cloud provider = gcp be added multiple times?

Reply:
No, as per the documentation, `AttributesBuilder` a key value store so worst case it would overwrite the cloud provider field.